django/test/runner.py

```diff
@@ -997,6 +1010,11 @@ def teardown_test_environment(self, **kwargs):
         unittest.removeHandler()
         teardown_test_environment()
 
+        if not self.keep_media and hasattr(settings, "TEST_MEDIA_ROOT"):
+            shutil.rmtree(settings.TEST_MEDIA_ROOT, ignore_errors=True)
```

Comment:
Generally speaking, the code that handles this should be part of `teardown_test_environment()`, IMO